provider/postgis/util.go

```diff
@@ -169,24 +156,64 @@ func replaceTokens(sql string, lyr *Layer, tile provider.Tile, withBuffer bool)
 	// replace query string tokens
 	z, x, y := tile.ZXY()
 	tokenReplacer := strings.NewReplacer(
-		bboxToken, bbox,
-		zoomToken, strconv.FormatUint(uint64(z), 10),
-		zToken, strconv.FormatUint(uint64(z), 10),
-		xToken, strconv.FormatUint(uint64(x), 10),
-		yToken, strconv.FormatUint(uint64(y), 10),
-		idFieldToken, lyr.IDFieldName(),
-		geomFieldToken, lyr.GeomFieldName(),
-		geomTypeToken, geoType,
-		scaleDenominatorToken, strconv.FormatFloat(scaleDenominator, 'f', -1, 64),
-		pixelWidthToken, strconv.FormatFloat(pixelWidth, 'f', -1, 64),
-		pixelHeightToken, strconv.FormatFloat(pixelHeight, 'f', -1, 64),
+		config.BboxToken, bbox,
+		config.ZoomToken, strconv.FormatUint(uint64(z), 10),
+		config.ZToken, strconv.FormatUint(uint64(z), 10),
+		config.XToken, strconv.FormatUint(uint64(x), 10),
+		config.YToken, strconv.FormatUint(uint64(y), 10),
+		config.ScaleDenominatorToken, strconv.FormatFloat(scaleDenominator, 'f', -1, 64),
+		config.PixelWidthToken, strconv.FormatFloat(pixelWidth, 'f', -1, 64),
+		config.PixelHeightToken, strconv.FormatFloat(pixelHeight, 'f', -1, 64),
+		config.IdFieldToken, lyr.IDFieldName(),
+		config.GeomFieldToken, lyr.GeomFieldName(),
+		config.GeomTypeToken, geoType,
 	)
 
 	uppercaseTokenSQL := uppercaseTokens(sql)
 
 	return tokenReplacer.Replace(uppercaseTokenSQL), nil
 }
 
+// replaceParams substitutes configured query parameter tokens for their values
+// within the provided SQL string
+func replaceParams(ctx context.Context, sql string) (string, error) {
+	if ctx.Value("params") == nil {
```

Comment:
maybe make the value "params" a const since you're referencing it multiple times.

Reply:
After reading the commit, I'm not sure if passing the params around via the context is the best approach. I know this allows us to avoid updating the provider interfaces, but I'm generally not a fan of stashing values into context.

cc: @gdey @ear7h